plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -341,6 +341,21 @@ public PostgreSqlClient(
                         .build());
     }
 
+    private Optional<String> typedCast(Type type)
+    {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            return varcharType.getLength().map(length -> format("varchar(%d)", length)).or(() -> Optional.of("varchar"));
```

Comment:
Trino bounded varchar can be of length 2147483646, PostgreSQL has a different limit